homeassistant/components/vesync/humidifier.py

```diff
@@ -96,6 +96,15 @@ def _get_vs_mode(ha_mode: str) -> str | None:
     return HA_TO_VS_MODE_MAP.get(ha_mode)
 
 
+def _get_available_modes(mist_modes: list[str]) -> list[str]:
+    """Return the available mist modes."""
+    return [
+        ha_mode
+        for ha_mode in (_get_ha_mode(vs_mode) for vs_mode in mist_modes)
+        if ha_mode
+    ]
```

Comment:
why?

Reply:
I want to test the function separately from device.

Reply:
We should still mock a device, as we don't want to mock internals

Reply:
The condition cannot be replicated by mocking a device.  `available_modes` is calculated based on `device.mist_modes` which is a hard coded in [pyvesync](https://github.com/webdjoe/pyvesync/blob/535d5324727546037f9d37bce4dbed4358e501a5/src/pyvesync/vesyncfan.py#L22). I think mocking something internal to pyvesync would be worse.

Reply:
Rewrote the test with mocked humidifier device.